service/history/commandChecker.go

```diff
@@ -173,6 +172,57 @@ func (c *workflowSizeChecker) checkIfMemoSizeExceedsLimit(
 	return nil
 }
 
+type childWorkflowLimitExceededErr string
+
+func (err childWorkflowLimitExceededErr) Error() string {
+	return strings.ToLower(string(err))
+}
+
+func (err childWorkflowLimitExceededErr) UserMessage() string {
+	return string(err)
+}
+
+func (c *workflowSizeChecker) checkIfNumChildWorkflowsExceedsLimit() error {
```

Comment:
Using this, the workflow will be failed. I am not sure if this is the behavior we plan to. I think it would be good just failed the workflow task